sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobApiTests.java

```diff
@@ -1797,6 +1800,50 @@ public void uploadAndDownloadRegionLengthWithDiffBlobClients(int regionLength, i
         assertArraysEqual(data.array(), plaintextOut.toByteArray());
     }
 
+    @Test
+    public void validateContentEncryptionKeyProtocol() throws IOException {
+        bec = getEncryptionClient(EncryptionVersion.V2_1, generateBlobName());
+        bec.uploadWithResponse(new BlobParallelUploadOptions(DATA.getDefaultInputStream()), null, null);
+        Response<BlobProperties> response = bec.getPropertiesWithResponse(null, null, null);
+        String encryptionMetadata = response.getHeaders().get(X_MS_META_ENCRYPTIONDATA).getValue();
+        EncryptionData encryptionData;
+        try (JsonReader jsonReader = JsonProviders.createReader(encryptionMetadata)) {
+            encryptionData = EncryptionData.fromJson(jsonReader);
+        }
+        byte[] cek = fakeKey.unwrapKey(encryptionData.getWrappedContentKey().getAlgorithm(),
+            encryptionData.getWrappedContentKey().getEncryptedKey()).block();
+
+        ByteArrayInputStream keyStream = new ByteArrayInputStream(cek);
+        byte[] protocolBytes = new byte[3];
+        keyStream.read(protocolBytes);
+
+        assertEquals(0, ByteBuffer.wrap(encryptionData.getEncryptionAgent().getProtocol().getBytes(StandardCharsets.UTF_8))
```

Comment:
Then instead of matching based off of what is in the metadata, we should declare the expected encryption protocol string in the parameterization and make sure that we are asserting against that value. This will help us definitively assert that it is the correct value instead of using a derived value for the assertion (that could have a bug in it).